shimTool/guiUtils.py

```diff
@@ -24,6 +29,93 @@
+        self.min_val = min_val
+        self.max_val = max_val
+        self.num_ticks = 6  # Number of tick marks on the color bar
+        self.tick_items = []  # To store tick mark items
+        self.label_items = []  # To store label items
+
+    def boundingRect(self):
+        return QRectF(0, 0, 50, 475)  # Width and height of the color bar
+
+    def paint(self, painter, option, widget):
+        rect = self.boundingRect()
+        gradient_rect = QRectF(10, 0, 40, 475)  # Adjust the gradient rectangle to be slightly smaller
+
+        # Define the gradient
+        gradient = QLinearGradient(
+            gradient_rect.left(), gradient_rect.top(), gradient_rect.left(), gradient_rect.bottom()
+        )
+        gradient.setColorAt(1, QColor(0, 0, 0))  # black
+        gradient.setColorAt(0, QColor(255, 255, 255))  # white
+
+        painter.fillRect(gradient_rect, gradient)
+
+        # Calculate positions for numerical labels
+        tick_positions = [rect.top() + i * (rect.height() / (self.num_ticks - 1)) for i in range(self.num_ticks)]
+
+        # Draw tick marks and labels
+        font = QFont()
+        font.setPixelSize(8)  # Adjust font size as needed
+        painter.setFont(font)
+
+        pen = QPen(QColor(0, 0, 0))  # Pen for tick marks
+        painter.setPen(pen)
+
+        # Clear previous tick and label items
+        for item in self.tick_items + self.label_items:
+            if item.scene():
+                item.scene().removeItem(item)
+        self.tick_items.clear()
+        self.label_items.clear()
+
+        for pos in tick_positions:
+
+            tick_line = QGraphicsLineItem(
+                round(gradient_rect.left()) - 10, round(pos), round(gradient_rect.left()) - 5, round(pos), self
+            )
+            self.tick_items.append(tick_line)
+            # self.scene().addItem(tick_line)  # Add tick mark to the scene
+
+            value = self.max_val - ((pos - rect.top()) / rect.height()) * (self.max_val - self.min_val)
+            tick_text = QGraphicsTextItem(f"{value:.2f}", self)
+            tick_text.setFont(font)
+            tick_text.setDefaultTextColor(QColor(0, 0, 0))
+            tick_text.setPos(round(gradient_rect.left()) - 45, round(pos) - 8)
+            self.label_items.append(tick_text)
+            # self.scene().addItem(tick_text)  # Add label to the scene
+
+
+class ColorBar(QGraphicsView):
+    def __init__(self, parent=None):
+        super(ColorBar, self).__init__(parent)
+        self.scene = QGraphicsScene(self)
+        self.setScene(self.scene)
+        self.colorbar_item = None
+        self.update_colorbar(np.array([]))
+
+    def update_colorbar(self, data):
+        self.scene.clear()
+        if data.size == 0:
+            return
+        min_val = np.nanmin(data)
+        max_val = np.nanmax(data)
+        self.colorbar_item = ColorBarItem(min_val, max_val)
+        self.scene.addItem(self.colorbar_item)
+
+        # fig, ax = plt.subplots(figsize=(1, 5))
```

Comment:
same thing i will stick to it too once i clean up the code this week. but wanna just remove commented out code before merging them in